src/axom/primal/geometry/BoundingBox.hpp

```diff
@@ -435,21 +437,19 @@ template < typename OtherType  >
 bool BoundingBox< T,NDIMS >::intersectsWith(
   const BoundingBox< OtherType, NDIMS >& otherBB ) const
 {
+  bool status = true;
+
   // AABBs cannot intersect if they are separated along any dimension
   for ( int i=0 ; i < NDIMS ; ++i )
   {
 
-    if ( (m_max[ i ] < otherBB.m_min[ i ]) ||
-         (m_min[ i ] > otherBB.m_max[ i ]) )
-    {
-
-      return false;
-
-    }  // END if
-
+    status = status && detail::intersect_bbox_bbox_test( m_min[ i ],
```

Comment:
It might not matter, but this function *might* be more efficient in its original form:
E.g.
```cpp
 for( ... )
 {
  if(detail::intersect_bbox_bbox_test( m_min[ i ], m_max[ i ], otherBB.m_min[ i ], otherBB.m_max[ i ] ) )
     return false;
 }
  return true;
```
I suppose we'd need to profile it to find out, (which is probably not worth spending time on)...

Reply:
I would expect the compiler to inline this function. I agree. It's not worth spending time on this...